src/main/java/io/spokestack/spokestack/tts/TTSManager.java

```diff
@@ -236,6 +249,31 @@ private void raiseError(Throwable e) {
         }
     }
 
+    @Override
+    public void eventReceived(@NonNull TTSEvent event) {
+        switch (event.type) {
+            case AUDIO_AVAILABLE:
+            case ERROR:
+                this.synthesizing = false;
+                processQueue();
+            default:
+                break;
+        }
+    }
+
+    private void processQueue() {
+        SynthesisRequest request = null;
+        synchronized (lock) {
+            if (!this.synthesizing) {
```

Comment:
there is a race here where two threads submit synthesis requests and both notice that `this.synthesizing` is false, so they both submit requests concurrently, so the write to the `synthesizing` variable will need to be protected by the lock
you can also do a double-checked locking optimization around this synchronized block

Reply:
Sigh, yep. I _knew_ I really wanted to be able to synchronize on `synthesizing`.

Reply:
could just rename the whole class the synthronizer

Reply:
don't tempt me